pkg/kubelet/images/image_manager.go

```diff
@@ -85,7 +87,15 @@ func (m *imageManager) EnsureImageExists(pod *api.Pod, container *api.Container,
 		glog.Errorf("Couldn't make a ref to pod %v, container %v: '%v'", pod.Name, container.Name, err)
 	}
 
-	spec := kubecontainer.ImageSpec{Image: container.Image}
+	// If the image contains no tag or digest, a default tag should be applied.
+	image, err := applyDefaultImageTag(container.Image)
```

Comment:
The bottom line is that we don't know if or what default is applied in different runtimes, so it's better if kubelet applies the tag. BTW, all the unit tests are broken. I'll look at them.